### PR TITLE
refactor: remove API v1 responses exception captures

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR remove all exception capture blocks for API v1 responses handler.

Refs #4871 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Improving and passing existent tests.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/src/argilla_server/bulk/records_bulk.py [hunk 1]
```diff
@@ -99,10 +100,10 @@ async def _upsert_records_suggestions(
                     try:
                         SuggestionCreateValidator(suggestion_create).validate_for(question.parsed_settings, record)
                         upsert_many_suggestions.append(dict(**suggestion_create.dict(), record_id=record.id))
-                    except ValueError as ex:
+                    except (UnprocessableEntityError, ValueError) as ex:
```
Comment (jfcalvo) [inline argilla-server/src/argilla_server/bulk/records_bulk.py:103]: @frascuchon once that everything is refactored I will change these kind of lines to be something like:

```suggestion
                    except UnprocessableEntityError as ex:
```

Comment (frascuchon) [reply argilla-server/src/argilla_server/bulk/records_bulk.py:103]: Okay but the raised error shouldn't be UnprocessableEntity?

File: argilla-server/src/argilla_server/bulk/records_bulk.py [hunk 2]
```diff
@@ -131,7 +132,7 @@ async def _upsert_records_responses(
 
                     ResponseCreateValidator(response_create).validate_for(record)
                     upsert_many_responses.append(dict(**response_create.dict(), record_id=record.id))
-            except ValueError as ex:
+            except (UnprocessableEntityError, ValueError) as ex:
                 raise ValueError(f"Record at position {idx} does not have valid responses because {ex}") from ex
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/bulk/records_bulk.py:136]: Same here. ValueError is not handled

File: argilla-server/src/argilla_server/contexts/datasets.py [hunk 3]
```diff
@@ -696,7 +694,7 @@ async def _build_record_suggestions(
                 )
             )
 
-        except ValueError as e:
+        except (UnprocessableEntityError, ValueError) as e:
             raise ValueError(f"suggestion for question_id={suggestion_create.question_id} is not valid: {e}") from e
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/contexts/datasets.py:698]: Same here

**Review by frascuchon — APPROVED** [review]

File: argilla-server/src/argilla_server/bulk/records_bulk.py (hunk 1 above)
Comment (jfcalvo) [reply argilla-server/src/argilla_server/bulk/records_bulk.py:103]: Yes, it's changed to raise `UnprocessableEntityError` in another PR: https://github.com/argilla-io/argilla/pull/4912/files#diff-8b6a4a23212d32e0db897e0fd54b482d2246be5f98c24915ce4f93923f14e8d5R107

**Pull request merged**